sbi/inference/trainers/vfpe/base_vf_inference.py

```diff
@@ -632,3 +659,798 @@ def _loss(
+                or a callable that implements the `MaskedVectorFieldEstimatorBuilder`
+                protocol with `__call__` that receives `inputs`, `condition_mask`,
+                and `edge_mask` and returns a `MaskedConditionalVectorFieldEstimator`.
+            device: Device to run the training on.
+            logging_level: Logging level for the training. Can be an integer or a
+                string.
+            summary_writer: Tensorboard summary writer.
+            show_progress_bars: Whether to show progress bars during training.
+            kwargs: Additional keyword arguments passed to the default builder if
+                `vector_field_estimator_builder` is a string.
+        """
+
+        super().__init__(
+            prior=prior,
+            device=device,
+            logging_level=logging_level,
+            summary_writer=summary_writer,
+            show_progress_bars=show_progress_bars,
+        )
+
+        # As detailed in the docstring, `vector_field_estimator` is either a string or
+        # a callable. The function creating the neural network is attached to
+        # `_build_neural_net`. It will be called in the first round and receive
+        # thetas and xs as inputs, so that they can be used for shape inference and
+        # potentially for z-scoring.
+        check_estimator_arg(mvf_estimator_builder)
+        if isinstance(mvf_estimator_builder, str):
+            self._build_neural_net = self._build_default_nn_fn(
+                vector_field_estimator_builder=mvf_estimator_builder,
+                **kwargs,
+            )
+        else:
+            self._build_neural_net = mvf_estimator_builder
+
+        self._proposal_roundwise = []
+
+        self.latent_idx = (
+            torch.as_tensor(latent_idx, dtype=torch.long)
+            if latent_idx is not None
+            else None
+        )
+        self.observed_idx = (
+            torch.as_tensor(observed_idx, dtype=torch.long)
+            if observed_idx is not None
+            else None
+        )
+
+    @abstractmethod
+    def _build_default_nn_fn(self, **kwargs) -> MaskedVectorFieldEstimatorBuilder:
+        pass
+
+    # TODO: Introduced to avoid conflicts, should adjust
+    def _get_potential_function(
+        self, prior: Distribution, estimator: ConditionalVectorFieldEstimator
+    ) -> Tuple[VectorFieldBasedPotential, TorchTransform]:
+        r"""Gets the potential function gradient for vector field estimators.
+
+        Args:
+            prior: The prior distribution.
+            estimator: The neural network modelling the vector field.
+        Returns:
+            The potential function and a transformation that maps
+            to unconstrained space.
+        """
+        potential_fn, theta_transform = vector_field_estimator_based_potential(
+            estimator,
+            prior,
+            x_o=None,
+        )
+        return potential_fn, theta_transform
+
+    def append_simulations(
```

Comment:
I think its great that one can specify condition mask and edge mask here. However I would not tie them to a "input". 

One can do it as follows: 
- We can let the user pass the "possible" (or wanted) condtion_mask and edge_masks and then randomly draw for each batch (N,) of these "possible" ones. (we might then also let the user pass "weights" on which he wants to draw more often).
- We can let the user pass a function that randomly generates a condition_mask and a function that generates an appropriate edge_mask given this condition mask.
- We can support both (the first approach because a bit tedious on higher dimensions).

So I think for now let's first do the first one.